### DISCUSSION TITLE
Scope 5.1 Developments

### DISCUSSION THREAD

**Discussion by sanyapawah** [Sunbird-coKreat/Community#22] [General Discussions]
This is the finalised 5.1 scope for Prashnavali including critical use cases and bugs.

> 

5.1 Items
--
Use Case : Admin controlled reminders/nudges sent to creators & reviewers -[PRD](https://project-sunbird.atlassian.net/wiki/spaces/PRD/pages/3197403157/Users+receive+messages+based+on+their+progress+on+Prashnavali) | [Github](https://github.com/Sunbird-coKreat/Community/discussions/15) | [JIRA](https://project-sunbird.atlassian.net/browse/CO-16)
Use Case : Question type column to be added in QuML 1.0 - [JIRA](https://project-sunbird.atlassian.net/browse/IQ-114)

Use Case : Bulk Approve button to be added in QuML 1.0 - [JIRA](https://project-sunbird.atlassian.net/browse/IQ-115)

Use Case : Print preview: Document generated as printout should be easily editable - [PRD](https://docs.google.com/document/d/18V02h7eOtmKO-mEHDhpGPBnqk0N1DF4QXWXLIJB1vc4/edit)

Bug : Content type under My Projects to be corrected for Q.set - [JIRA](https://project-sunbird.atlassian.net/browse/IQ-116)

Bug : Bulk approval not working for Questions added from Library - QuML 0.5 - [JIRA](https://project-sunbird.atlassian.net/browse/IQ-117)

UI/UX enhancements on current use cases/features

Bug : Duplicate delay handling - [JIRA](https://project-sunbird.atlassian.net/browse/CO-92)

Bug : Send back for corrections button not working correctly - [JIRA](https://project-sunbird.atlassian.net/browse/IQ-118)

Use Case : Send back for corrections button to be removed for questions added from Library - [JIRA](https://project-sunbird.atlassian.net/browse/IQ-119)

CC : @surendrasinghs @pallakartheekreddy 
Please review the approach for the above finalised scope. We will commence the developments accordingly.

**Comment by tarang27** [comment]
Hi @pallakartheekreddy @surendrasinghs please review the above functional & tech approach and let us know if any changes need to be done. We would also need your help identifying any other building blocks that would be used in the above developments.

Thanks

**Comment by surendrasinghs** [comment]
@sanyapawah @tarang27 Thank you for sharing this. Please see my responses in-line.

> Use Case : Admin controlled reminders/nudges sent to creators & reviewers -[PRD](https://project-sunbird.atlassian.net/wiki/spaces/PRD/pages/3197403157/Users+receive+messages+based+on+their+progress+on+Prashnavali) | [Github](https://github.com/Sunbird-coKreat/Community/discussions/15) | [JIRA](https://project-sunbird.atlassian.net/browse/CO-16) Use Case : Question type column to be added in QuML 1.0 - [JIRA](https://project-sunbird.atlassian.net/browse/IQ-114)

Commented on the Confluence.

> Use Case : Bulk Approve button to be added in QuML 1.0 - [JIRA](https://project-sunbird.atlassian.net/browse/IQ-115)

Please share the implementation details. Commented on the JIRA ticket for the same.

> Use Case : Publish Question set for consumption on DIKSHA - PRD

I believe this is a solution & tech design item and no development is planned for this.

> Use Case : DIKSHA Vani Bot usage for grievance redressal

I believe this is a solution design item and no development is planned for this.

> Use Case : Print preview: Document generated as printout should be easily editable - [PRD](https://docs.google.com/document/d/18V02h7eOtmKO-mEHDhpGPBnqk0N1DF4QXWXLIJB1vc4/edit)

Please share the implementation details. Commented on the PRD/Google Doc for the same.

> Bug : Content type under My Projects to be corrected for Q.set - [JIRA](https://project-sunbird.atlassian.net/browse/IQ-116)

Please share the implementation details. Commented on the JIRA ticket for the same.

> Bug : Bulk approval not working for Questions added from Library - QuML 0.5 - [JIRA](https://project-sunbird.atlassian.net/browse/IQ-117)

Please share the implementation details. Commented on the JIRA ticket for the same.

> UI/UX enhancements on current use cases/features

Are these defined already? Please plan them as per the release calendar.

> Bug : Duplicate delay handling - [JIRA](https://project-sunbird.atlassian.net/browse/CO-92)

Please share the implementation details. Commented on the JIRA ticket for the same.

> Bug : Send back for corrections button not working correctly - [JIRA](https://project-sunbird.atlassian.net/browse/IQ-118)

Please share the implementation details. Commented on the JIRA ticket for the same.

> Use Case : Send back for corrections button to be removed for questions added from Library - [JIRA](https://project-sunbird.atlassian.net/browse/IQ-119)

Looks good.

### Few general observations and requests
1. Eventually let's have all items in the JIRA. Please add if any.
2. Please update the following in the JIRA ticket: Contributor details, Sprint, and Priority.
3. Please share a status update on release items at the end of each week in the sprint. It helps anticipate changes in planned scope.

**Comment by surendrasinghs** [comment]
Can you please plan to resolve bugs and defects from [this list of tickets](https://project-sunbird.atlassian.net/issues/?jql=(assignee%20in%20(61dc64a849f19500698bad0c%2C%20557058%3A6730c54d-a5d3-45bd-8466-7af7af109508%2C%20630473b96acf9eeb443dc480%2C%2063231fa13ead8137f44ecb83%2C%2070121%3Adf8e4b2b-90cf-4940-b44d-e5190e292f29%2C%205f742ecc25fbdf00682ca726%2C%20603c9d6cc668f4006aeaa6e1)%20or%20%22Contributor%5BDropdown%5D%22%20in%20(Samagra)%20or%20%22Contributor%20Type%5BSelect%20List%20(cascading)%5D%22%20in%20(Samagra))%20AND%20status%20not%20in%20(closed)%20and%20issuetype%20not%20in%20(Checklist-Item)%20ORDER%20BY%20issuetype%20asc%2C%20created%20DESC)? I have filtered for relevant tickets related to your contributions.

**Comment by sanyapawah** [comment]
@surendrasinghs @pallakartheekreddy @tarang27 adding the latest discussed approaches discussed for the above listed 5.1 scope use cases and bugs :

Use Case : Admin controlled reminders/nudges sent to creators & reviewers -[PRD](https://project-sunbird.atlassian.net/wiki/spaces/PRD/pages/3197403157/Users+receive+messages+based+on+their+progress+on+Prashnavali) | https://github.com/Sunbird-coKreat/Community/discussions/15 | [JIRA](https://project-sunbird.atlassian.net/browse/CO-16)
Approach : A checkbox will be added on the project creation page for the admin to enable reminders/nudges to be sent to the creators and reviewers. Depending upon the system environment variable, the checkbox will be blank in the beginning. The API can expect a maximum of 1000 requests at a time. Template for the sms/mail reminder to be developed. 

Bug: Send back for corrections button not working correctly - [JIRA](https://project-sunbird.atlassian.net/browse/IQ-118)
Approach : Presently when the sourcing org reviewer is sending back for corrections, the contributor is seeing the status of question as "Live". To resolve this, an edit mode has to be added in the request for the API to correct the functionality.

**Comment by sanyapawah** [comment]
Hi @surendrasinghs @pallakartheekreddy 
Have added the Sprint versions to the open JIRA tickets for your reference.

**Comment by sanyapawah** [comment]
Signed off in CoKreat Rel 5.1.0
Use Case : Admin controlled reminders/nudges sent to creators & reviewers -[PRD](https://project-sunbird.atlassian.net/wiki/spaces/PRD/pages/3197403157/Users+receive+messages+based+on+their+progress+on+Prashnavali) | https://github.com/Sunbird-coKreat/Community/discussions/15 | [JIRA](https://project-sunbird.atlassian.net/browse/CO-16)

Signed off in CoKreat Rel 5.0.3
Use Case : Question type column to be added in QuML 1.0 - [JIRA](https://project-sunbird.atlassian.net/browse/IQ-114)

Use Case : Bulk Approve button to be added in QuML 1.0 - [JIRA](https://project-sunbird.atlassian.net/browse/IQ-115)

Use Case : Print preview: Document generated as printout should be easily editable - [PRD](https://docs.google.com/document/d/18V02h7eOtmKO-mEHDhpGPBnqk0N1DF4QXWXLIJB1vc4/edit)

Bug : Content type under My Projects to be corrected for Q.set - [JIRA](https://project-sunbird.atlassian.net/browse/IQ-116)

Bug : Bulk approval not working for Questions added from Library - QuML 0.5 - [JIRA](https://project-sunbird.atlassian.net/browse/IQ-117)

UI/UX enhancements on current use cases/features

Bug : Duplicate delay handling - [JIRA](https://project-sunbird.atlassian.net/browse/CO-92)

Bug : Send back for corrections button not working correctly - [JIRA](https://project-sunbird.atlassian.net/browse/IQ-118)

Use Case : Send back for corrections button to be removed for questions added from Library - [JIRA](https://project-sunbird.atlassian.net/browse/IQ-119)

**Comment by sanyapawah** [comment]
**Signed off : CoKreat Rel 5.1**
Use Case : Admin controlled reminders/nudges sent to creators & reviewers -[PRD](https://project-sunbird.atlassian.net/wiki/spaces/PRD/pages/3197403157/Users+receive+messages+based+on+their+progress+on+Prashnavali) | https://github.com/Sunbird-coKreat/Community/discussions/15 | [JIRA](https://project-sunbird.atlassian.net/browse/CO-16)

**Signed off : CoKreat Rel 5.0.3**
Use Case : Question type column to be added in QuML 1.0 - [JIRA](https://project-sunbird.atlassian.net/browse/IQ-114)

**In progress**
Use Case : Bulk Approve button to be added in QuML 1.0 - [JIRA](https://project-sunbird.atlassian.net/browse/IQ-115)

**In progress**
Use Case : Print preview: Document generated as printout should be easily editable - [PRD](https://docs.google.com/document/d/18V02h7eOtmKO-mEHDhpGPBnqk0N1DF4QXWXLIJB1vc4/edit)

**Signed off : CoKreat Rel 5.0.3**
Bug : Content type under My Projects to be corrected for Q.set - [JIRA](https://project-sunbird.atlassian.net/browse/IQ-116)

**Closed**
Bug : Bulk approval not working for Questions added from Library - QuML 0.5 - [JIRA](https://project-sunbird.atlassian.net/browse/IQ-117)

**Signed off : CoKreat Rel 5.0.3**
Bug : Duplicate delay handling - [JIRA](https://project-sunbird.atlassian.net/browse/CO-92)

**In progress**
Use Case : Send back for corrections button to be removed for questions added from Library - [JIRA](https://project-sunbird.atlassian.net/browse/IQ-119)
